simplified version code exhibit behavior troubling testfn print print triggering result true return return testfn triggering result true intuitively bigger evaluate nd argument exit testfn returning however although nd argument statement getting evaluated function exit hit return kind guessed might problem return exiting seems confirmed testfn print print triggering result true return print tostring return triggering result true indeed happening make code actually return exit testfn